test/Operators/hybrid/extruded_cuda.jl

```diff
@@ -0,0 +1,68 @@
+#=
+julia --project
+using Revise; include(joinpath("test", "Spaces", "extruded_cuda.jl"))
+=#
+using LinearAlgebra, IntervalSets, UnPack
+using ClimaComms
+using CUDA
+using ClimaComms: SingletonCommsContext
+import ClimaCore:
+    Domains, Topologies, Meshes, Spaces, Geometry, column, Fields, Operators
+using Test
+
+function get_space(context)
+    FT = Float64
+
+    # Define vert domain and mesh
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(0.0),
+        Geometry.ZPoint{FT}(1.0);
+        boundary_tags = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = 10)
+
+    # Define vert topology and space
+    verttopology = Topologies.IntervalTopology(context, vertmesh)
+    vert_center_space = Spaces.CenterFiniteDifferenceSpace(verttopology)
+
+    # Define horz domain and mesh
+    horzdomain = Domains.SphereDomain(FT(30.0))
+    horzmesh = Meshes.EquiangularCubedSphere(horzdomain, 4)
+
+    quad = Spaces.Quadratures.GLL{3 + 1}()
+
+    # Define horz topology and space
+    horztopology = Topologies.Topology2D(context, horzmesh)
+    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
+
+    # Define hv space
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+    return hv_center_space
+end
+
+@testset "GPU extruded sphere divergence" begin
+    cpu_context = SingletonCommsContext(ClimaComms.CPUDevice())
+    gpu_context = SingletonCommsContext(ClimaComms.CUDADevice())
+    device = ClimaComms.device() #ClimaComms.CUDADevice()
+    println("running test on $device device")
+
+
+    # Define hv CPU space
+    hv_center_space_cpu = get_space(cpu_context)
+
+    # Define hv GPU space
+    hv_center_space_gpu = get_space(gpu_context)
+
+    # Define CPU & GPU coordinate fields
+    coords_cpu = Fields.coordinate_field(hv_center_space_cpu)
+    coords_gpu = Fields.coordinate_field(hv_center_space_gpu)
+
+    # Define CPU & GPU fields
+    x_cpu = Geometry.UVWVector.(cosd.(coords_cpu.lat), 0.0, 0.0)
+    x_gpu = Geometry.UVWVector.(cosd.(coords_gpu.lat), 0.0, 0.0)
+
+    # Test div operator
+    div = Operators.Divergence()
+    @test parent(div.(x_cpu)) ≈ Array(parent(div.(x_gpu)))
```

Comment:
Can we test other operators (`grad`, `curl`), and some composed operators (`div.(grad.(x))`)?

Reply:
Yes, absolutely. I had left a [comment](https://github.com/CliMA/ClimaCore.jl/issues/1169#issuecomment-1535552647) in the related Issue